microsoft/testsuites/network/networksettings.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.tools import Ethtool
+from lisa.util import LisaException, SkippedException, UnsupportedOperationException
+
+
+@TestSuiteMetadata(
+    area="network",
+    category="functional",
+    description="""
+    This test suite runs the ethtool related network test cases.
+    """,
+)
+class NetworkSettings(TestSuite):
+    @TestCaseMetadata(
+        description="""
+            This test case verifies if ring buffer settings can be changed with ethtool.
+
+            Steps:
+            1. Get the current ring buffer settings.
+            2. Change the rx and tx value to new_values using ethtool.
+            3. Get the settings again and validate the current rx and tx
+                values are equal to the new_values assigned.
+            4. Revert back the rx and tx value to their orginal values.
+
+        """,
+        priority=1,
+    )
+    def validate_ringbuffer_settings_change(self, node: Node) -> None:
+        ethtool = node.tools[Ethtool]
+        try:
+            devices_settings = ethtool.get_all_device_ring_buffer_settings()
+        except UnsupportedOperationException:
+            raise SkippedException
+        except Exception:
```

Comment:
It doesn't need to rewrite exception here. If it's needed, it should include the raw exception, it's important message about why it's failed.